streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -152,6 +152,24 @@ boolean isRepartitionTopic(final String topic) {
         return repartitionTopics.contains(topic);
     }
 
+    public boolean hasPersistentLocalStore() {
+        for (final StateStore store : stateStores) {
+            if (store.persistent()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean hasPersistentGlobalStore() {
```

Comment:
As above: we should add a test for this method

Reply:
test case added.